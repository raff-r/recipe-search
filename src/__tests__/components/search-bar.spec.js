import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';

import SearchBar from '../../components/search-bar';

describe('SearchBar Component', () => {

    const mockStore = configureStore([thunk]);
    const initialState = {};
    let component, store;

  beforeEach(() => {
    store = mockStore(initialState);
    component = mount( <SearchBar store={store}/> );
  });

  it('should match the snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('entering text into form and submit', () => {

      let input;

      beforeEach(() => {
          input = component.find('input#searchTerm');
          input.node.value = 'My new value';
          input.simulate('change');
      });

      it('shows that text in the search term input', () => {
          expect(input.node.value).toEqual('My new value');
      });

      it('when submitted, sets has-result class', () => {
          component.simulate('submit');
          expect(component.html()).toContain('has-results');
      });
  });

});