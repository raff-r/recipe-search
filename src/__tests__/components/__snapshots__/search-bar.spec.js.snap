// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should match the snapshot 1`] = `
<Connect(SearchBar)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SearchBar
    searchApi={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <div
          className="row"
        >
          <div
            className="col-xs-12 col-md-10"
          >
            <div
              className="row"
            >
              <div
                className="col-xs-12 col-md-4"
              >
                <label
                  htmlFor="searchTerm"
                >
                  Search term
                </label>
                <input
                  className="form-control"
                  id="searchTerm"
                  name="q"
                  onChange={[Function]}
                  placeholder="Search…"
                  type="text"
                />
              </div>
              <div
                className="col-xs-12 col-md-4"
              >
                <label
                  htmlFor="diet"
                >
                  Diet
                </label>
                <select
                  className="form-control"
                  id="diet"
                  name="diet"
                  onChange={[Function]}
                >
                  <option
                    value=""
                  >
                    Select option
                  </option>
                  <option
                    value="balanced"
                  >
                    Balanced
                  </option>
                  <option
                    value="high-protein"
                  >
                    High protein
                  </option>
                  <option
                    value="high-fiber"
                  >
                    High fiber
                  </option>
                  <option
                    value="low-fat"
                  >
                    Loe fat
                  </option>
                  <option
                    value="low-carb"
                  >
                    Low carb
                  </option>
                  <option
                    value="low-sodium"
                  >
                    Low sodium
                  </option>
                </select>
              </div>
              <div
                className="col-xs-12 col-md-4"
              >
                <label
                  htmlFor="calories"
                >
                  Number of calories
                </label>
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                  >
                    <label>
                      <input
                        checked={false}
                        id="caloriesType-gte"
                        name="caloriesType"
                        onChange={[Function]}
                        type="radio"
                        value="gte"
                      />
                       &gt;
                    </label>
                  </span>
                  <span
                    className="input-group-addon"
                  >
                    <label>
                      <input
                        checked={false}
                        id="caloriesType-lte"
                        name="caloriesType"
                        onChange={[Function]}
                        type="radio"
                        value="lte"
                      />
                       &lt;
                    </label>
                  </span>
                  <input
                    className="form-control"
                    id="calories"
                    name="calories"
                    onChange={[Function]}
                    placeholder="Calories"
                    type="number"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-xs-12 col-md-2"
          >
            <button
              className="btn btn-default btn-block"
              type="submit"
            >
              <span
                className="glyphicon glyphicon-search"
              />
               Search
            </button>
          </div>
        </div>
      </div>
    </form>
  </SearchBar>
</Connect(SearchBar)>
`;
